std-bits/table/src/main/java/org/enso/table/data/column/operation/unary/SignumOperation.java

```diff
@@ -0,0 +1,114 @@
+
+  @Override
+  public String getName() {
+    return "signum";
+  }
+
+  @Override
+  public boolean canApply(ColumnStorage<?> storage) {
+    return switch (storage.getType()) {
+      case IntegerType ignored -> true;
+      case FloatType ignored -> true;
+      case BigIntegerType ignored -> true;
+      case BigDecimalType ignored -> true;
+      case NullType ignored -> true;
+      default -> false;
+    };
+  }
+
+  @Override
+  public ColumnStorage<?> apply(
+      ColumnStorage<?> storage, MapOperationProblemAggregator problemAggregator) {
+    if (storage.getType() instanceof NullType) {
+      return new NullStorage(Math.toIntExact(storage.getSize()));
+    }
+
+    if (storage instanceof ColumnLongStorage columnLongStorage) {
+      return StorageIterators.mapOverLongStorage(
+          columnLongStorage,
+          Builder.getForLong(IntegerType.INT_8, storage.getSize(), problemAggregator),
+          (index, value, isNothing) -> signum(value));
+    }
+
+    if (storage instanceof ColumnDoubleStorage columnStorage) {
+      return StorageIterators.mapOverStorage(
+          columnStorage,
+          Builder.getForDouble(FloatType.FLOAT_64, storage.getSize(), problemAggregator),
+          (index, value) -> signum(value));
+    }
+
+    if (storage.getType() instanceof BigIntegerType bigIntegerType) {
+      ColumnStorage<BigInteger> bigIntegerColumnStorage = bigIntegerType.asTypedStorage(storage);
+      return StorageIterators.mapOverStorage(
+          bigIntegerColumnStorage,
+          Builder.getForLong(IntegerType.INT_8, storage.getSize(), problemAggregator),
+          (index, value) -> signum(value));
+    }
+
+    if (storage.getType() instanceof BigDecimalType bigDecimalType) {
+      ColumnStorage<BigDecimal> bigDecimalColumnStorage = bigDecimalType.asTypedStorage(storage);
+      return StorageIterators.mapOverStorage(
+          bigDecimalColumnStorage,
+          Builder.getForLong(IntegerType.INT_8, storage.getSize(), problemAggregator),
+          (index, value) -> signum(value));
+    }
+
+    // Fallback for Mixed and any other storage implementations
+    var builder = Builder.getInferredBuilder(storage.getSize(), problemAggregator);
+    for (long i = 0; i < storage.getSize(); i++) {
+      if (storage.isNothing(i)) {
+        builder.appendNulls(1);
+      } else {
+        var item = storage.getItemBoxed(i);
+        switch (item) {
+          case Long lng -> builder.append(signum(lng));
+          case Double dbl -> builder.append(signum(dbl));
+          case BigInteger bi -> builder.append(signum(bi));
+          case BigDecimal bd -> builder.append(signum(bd));
+          default -> {
+            builder.appendNulls(1);
+            problemAggregator.reportIllegalArgumentError(
+                "Unsupported type for signum operation: " + item, Math.toIntExact(i));
+          }
+        }
+      }
+    }
+    return builder.seal();
+  }
+
+  private long signum(long x) {
+    return Long.compare(x, 0);
```

Comment:
Why not using https://docs.oracle.com/en/java/javase/21/docs/api/java.base/java/lang/Long.html#signum(long)?

Reply:
Good point, I somehow couldn't find this operation when I was looking so resorted to `compare` but that looks better.